nginx/nginx.conf

```diff
@@ -44,8 +44,8 @@ http {
 	# it could be "close" to close a keepalive connection
 	proxy_set_header Connection '';
 	proxy_set_header Host $host;
-	proxy_set_header X-Real-IP $remote_addr;
-	proxy_set_header X-Forwarded-For $remote_addr;
+	proxy_set_header X-Real-IP $proxy_add_x_forwarded_for;
```

Comment:
X-Real-IP would still be `$remote_addr`.

But in theory, we don't have enough information really to assume anything here. But you don't have a list of IPs in `X-Real-IP`.

In practice, you'd wanna use this: https://nginx.org/en/docs/http/ngx_http_realip_module.html

And pass along the correct IPs.

Reply:
Ftr I started with [this](https://github.com/getsentry/onpremise/issues/554#issuecomment-660522070). I don't `ag real-ip` anywhere in `sentry`. Rather, what I'm seeing is that `sentry` [logs](https://github.com/getsentry/sentry/blob/20.11.1/src/sentry/middleware/auth.py#L35) `request.META["REMOTE_ADDR"]`—which is [overriden](https://github.com/getsentry/sentry/blob/20.11.1/src/sentry/middleware/proxy.py#L97-L107) with `X-Forwarded-For` [by](https://github.com/getsentry/sentry/blob/20.11.1/src/sentry/middleware/proxy.py#L84) [default](https://github.com/getsentry/sentry/blob/20.11.1/src/sentry/conf/server.py#L1625)—and that's [where geo is knit in](https://github.com/getsentry/sentry/blob/20.11.1/src/sentry/models/userip.py#L37-L44). (`relay`, of course, geolocates the IP address provided in the request [body](https://docs.sentry.io/platforms/python/enriching-events/identify-user/), not the headers.)

Tweaked in a1716ebb2b8fd21c0ece58f3a0ac0989be398612.